bottomless/src/wal.rs

```diff
@@ -234,6 +253,47 @@ pub fn checksum_be(init: u64, page: &[u8]) -> u64 {
     ((s1 as u64) << 32) | (s2 as u64)
 }
 
+// 128KiB since many OSes already prefetch 128KiB and that's also unit of pricing in S3
+const COPY_BUF_SIZE: u64 = 128 * 1024;
+
+/// Copies up to `len` bytes from `src` to `dst`. Returns number of written bytes,
+/// which can differ from bytes read eg. when compression is used.
+async fn copy_range<R, W>(src: &mut R, dst: &mut W, len: u64) -> Result<u64>
+where
+    R: AsyncRead + Unpin,
+    W: AsyncWrite + Unpin,
+{
+    let capacity = COPY_BUF_SIZE.min(len) as usize;
+    let mut buf = Vec::with_capacity(capacity);
+    buf.spare_capacity_mut();
```

Comment:
I don't understand this idiom. Why is it needed to take a mutable reference to the uninitialized part, just to set the length later?

Reply:
It's advised by clippy - in general we don't need to initialise buffer every time, as we always require to fill it all. But clippy consider that a wrong operation and recommends filling it "by hand" using spare_capacity_mut.

Reply:
Got it, in that case I'd rather just silence clippy in this scope. But I suppose this code is going away anyway after Read::take()?

Reply:
There's another place using this method - page buffer used during restoration.